src/Swarm/Language/Pretty.hs

```diff
@@ -294,6 +294,8 @@ instance PrettyPrec TypeErr where
     "Record does not have a field with name" <+> pretty x <> ":" <+> ppr t
   prettyPrec _ (InvalidAtomic reason t) =
     "Invalid atomic block:" <+> ppr reason <> ":" <+> ppr t
+  prettyPrec _ Impredicative =
+    "Impredicative type encountered. See https://github.com/swarm-game/swarm/issues/351"
```

Comment:
Could we fit some more actionable info that would be useful even to people who don't know what impredicative types are?

Reply:
Well, not really, the problem is this is really just a band-aid on a bug.  We should have rejected this type before getting to this point, at some point when we would have been in a position to give a more reasonable error message.  But at the point where we find unification variables in a type it is kind of too late to give any useful info.  I guess I can at least make clear in the message that this is a bug in Swarm.